# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/test/mochitest/file_bug1742865_outer.sjs
# Commit: 9896c12c4907
# Full Hash: 9896c12c490709e214030cd99f598e1ffa0076de
# Author: Peter Van der Beken <peterv@propagandism.org>
# Date: 2021-12-17 03:51:53
# Regressor Bug: 1742865
# File Overlap Count: 1
# Description:
#   Bug 1742865 - Handle meta refresh correctly with session history in the parent. r=smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D132837
# ==============================================================================

diff -r 607f12999b71 -r 9896c12c4907 docshell/test/mochitest/file_bug1742865_outer.sjs
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/docshell/test/mochitest/file_bug1742865_outer.sjs	Thu Dec 16 15:31:56 2021 +0000
@@ -0,0 +1,25 @@
+Cu.importGlobalProperties(["URLSearchParams"]);
+
+function handleRequest(request, response) {
+  response.write(
+    `<!DOCTYPE html>
+<html>
+<head>
+  <meta charset="utf-8">
+  <script>
+    window.addEventListener("message", ({ data }) => {
+      if (data == "loadNext") {
+        location.href += "&loadnext=1";
+        return;
+      }
+      // Forward other messages to the frame.
+      document.getElementById("frame").contentWindow.postMessage(data, "*");
+    });
+  </script>
+</head>
+<body>
+  <iframe src="file_bug1742865.sjs?${request.queryString}" id="frame"></iframe>
+</body>
+</html>`
+  );
+}